x-pack/plugins/fleet/server/services/epm/archive/extract.ts

```diff
@@ -19,14 +19,23 @@ export async function untarBuffer(
 ): Promise<unknown> {
   const deflatedStream = bufferToStream(buffer);
   // use tar.list vs .extract to avoid writing to disk
-  const inflateStream = tar.list().on('entry', (entry: tar.FileStat) => {
+  const inflateStream = tar.list().on('entry', (entry) => {
+    // @ts-expect-error: Property 'header' does not exist on type 'ReadEntry'.
+    // `entry` is of type `tar.ReadEntry` and officially doesn't have a `header` property, but it does: https://github.com/npm/node-tar/blob/26a496e5fa74eeaa0c3539511560fc181ef56557/lib/read-entry.js#L15
+    // TODO: Should we use `entry.path` instead? Then we don't need the ts-expect-error (see: https://github.com/npm/node-tar/blob/26a496e5fa74eeaa0c3539511560fc181ef56557/lib/read-entry.js#L50)
     const path = entry.header.path || '';
     if (!filter({ path })) return;
+    // @ts-expect-error: Argument of type 'ReadEntry' is not assignable to parameter of type 'ReadableStream'.
+    // `entry` is of type `tar.ReadEntry` which inherits from `minipass` making it a `PassThrough` stream, so this should be ok
     streamToBuffer(entry).then((entryBuffer) => onEntry({ buffer: entryBuffer, path }));
   });
 
   return new Promise((resolve, reject) => {
-    inflateStream.on('end', resolve).on('error', reject);
+    inflateStream.on('end', () => {
+      resolve(undefined); // TODO: WTF why does TypeScript not allow us to call resolve without an argument??
+    });
+    // TODO: Can `inflateStream` truly never emit `error`? According to `@types/tar` it cannot
+    // inflateStream.on('error', reject);
```

Comment:
@elastic/fleet When upgrading to Node.js 18, I've had to upgrade the `tar` and the `@types/tar` package in order to be compatible with the newer version of Node.js. These new types unfortunately doesn't agree with your usage of `tar` in this file.

I'm not familiar enough with the logic in here to say what the best implementation is. Could I get you to look at the TODO's and changes to see if you agree or which changes to keep or change?

Reply:
I am not too familiar with this code either, but I see we have [integration tests](https://github.com/elastic/kibana/blob/main/x-pack/test/fleet_api_integration/apis/epm/file.ts#L117) that calls this logic to extract `tar.gz` files. 
So I think let's change it according to the recommendation from the `tar` module, and see if the tests are passing.